qutebrowser/browser/commands.py

```diff
@@ -479,7 +480,7 @@ def tab_give(self, win_id: int = None, keep: bool = False,
                                             "only one tab")
 
             tabbed_browser = self._new_tabbed_browser(
-                private=self._tabbed_browser.is_private)
+                private=private or self._tabbed_browser.is_private)
         else:
```

Comment:
You should check whether `--private` is set here and show an error - otherwise, someone could expect a tab to be private after `:tab-give --private 2`, but it's not.